src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < figures.length; i++) {
+            if (i < (figures.length / 2)) {
+                figures[i] = figureSupplier.getRandomFigure();
+                figures[i].toDrawFigure();
```

Comment:
It's a good practice to separate the logic of creating figures and drawing them. Consider moving the drawing logic outside of the loop after all figures have been created.